lib/src/domain/usecase/load_cast_usecase.dart

```diff
@@ -1,21 +1,36 @@
-import 'package:http/http.dart';
+import 'package:connectivity_plus/connectivity_plus.dart';
 
 import '../../core/usecase/i_usecase.dart';
 import '../../core/util/data_state.dart';
-import '../../data/repository/api_repository.dart';
+import '../../core/util/enums.dart';
+import '../../data/local/movie_database.dart';
 import '../entity/cast.dart';
 import '../entity/movie.dart';
 import '../repository/i_my_repository.dart';
 
-class LoadCastUseCase
-    extends IUseCase<Future<DataState<List<Cast>>>, Movie> {
+class LoadCastUseCase extends IUseCase<Future<DataState<List<Cast>>>, Movie> {
   final IMyRepository repository;
+  final MovieDatabase database;
 
-  LoadCastUseCase({IMyRepository? repository})
-      : repository = repository ?? APIRepository(client: Client());
+  LoadCastUseCase({
+    required this.repository,
+    required this.database,
+  });
 
   @override
   Future<DataState<List<Cast>>> call([Movie? params]) async {
-    return await repository.loadCast(params?.id);
+    if(await Connectivity().checkConnectivity() != ConnectivityResult.none){
+      final DataState<List<Cast>> apiResponse =
+      await repository.loadCast(params?.id);
+      if (apiResponse.state == Status.success) {
+        for(int i = 0; i < apiResponse.data!.length; i++){
+          final Cast cast = apiResponse.data![i];
+          await database.castDao.insertCast(cast);
+        }
+      }
```

Comment:
Instead to do this, you can made like a addAll, to call one by one the insert to every cast, this could help to made only one call to the database

Reply:
I did't know the database could accept all the data at once. Already fixed! 🚀